library/ssl_cli.c

```diff
@@ -1922,10 +1922,10 @@ static int ssl_parse_server_hello( mbedtls_ssl_context *ssl )
     }
 
     MBEDTLS_SSL_DEBUG_MSG( 3, ( "server hello, current time: %lu",
-                           ( (uint32_t) buf[2] << 24 ) |
-                           ( (uint32_t) buf[3] << 16 ) |
-                           ( (uint32_t) buf[4] <<  8 ) |
-                           ( (uint32_t) buf[5]       ) ) );
+                           (uint32_t)( (uint32_t) buf[2] << 24 ) |
```

Comment:
This is still wrong: the format is `%lu` but the argument type is `uint32_t`. Possible fixes:

* Make the format `%u` and leave the argument uncast. This is wrong on platforms where `unsigned int` is less than 32 bits, but Mbed TLS assumes that `unsigned int` is at least 32-bits anyway. It's obviously correct if `uint32_t` is an alias for `unsigned int`. It's also correct if `unsigned int` is 33 bits or more since the argument is promoted to `int` or `unsigned` and both are valid arguments to a `%u` specifier. I'm not sure if it's correct on hypothetical platforms where `uint32_t` and `unsigned` are both 32-bit types but they have different representations
* Keep the format as `%lu` and cast the argument to `unsigned long`.

Reply:
Just realised this when the warning finally threw on it. The original coverity bug was that the result of the shifts is unsigned int, rather than unsigned long, and I clearly misrread it (and this was before I realised we were not getting any warnings). Will fix in next patch.